Library/Homebrew/livecheck/strategy/gnu.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Homebrew::Livecheck::Strategy::Gnu
+  module_function
+
+  NICE_NAME = "GNU"
+
+  PROJECT_NAME_REGEXES = [
+    %r{/(?:software|gnu)/(.*?)/},
+    %r{//(.*?)\.gnu\.org(?:/)?$},
+  ].freeze
+  private_constant :PROJECT_NAME_REGEXES
+
+  SPECIAL_CASES = %w[
```

Comment:
Would be nice to have a test that verifies all these formulae still exist in homebrew-core.

Reply:
@samford correct me if I'm wrong, but wouldn't the `strategy` feature make `SPECIAL_CASES` obsolete? Wouldn't we just have to use `strategy :page_match` or something to prevent the GNU strategy from being used?

Reply:
Yes, for the `Gnu` and `Sourceforge` `SPECIAL_CASES` we can add/modify the `livecheck` block to use `strategy` instead. I was planning to address this after the Homebrew/brew migration but I can take care of the livecheckable changes if we want to do it as part of this process.

To be clear, the `Gnome` strategy’s `DEV_VERSION_ALLOWLIST` is different. This array constant is used in `#find_versions` to select the proper regex, instead of omitting the URL from matching the strategy.